src/Processors/QtiV2/Out/QuestionTypes/McqMapper.php

```diff
@@ -26,13 +30,32 @@ public function convert(BaseQuestionType $questionType, $interactionIdentifier,
         // Build <choiceInteraction>
         $valueIdentifierMap = [];
         $simpleChoiceCollection = new SimpleChoiceCollection();
+        $metadata = $question->get_metadata();
+
+        $feedbackOptions = [];
+        if (isset($metadata) && !empty($metadata->get_distractor_rationale_response_level())) {
+            foreach ($metadata->get_distractor_rationale_response_level() as $feed) {
+                $feedbackOptions[] = $feed;
+            }
+        }
+
+        $i = 0;
```

Comment:
Why not just use `$index` from inside the loop? Is `$index` a string key or an ordinal value? (if it's a string key then fair enough, but then `$index` should probably be called `$key` instead.)

Reply:
Done , sorry dint check $index . Now I have removed $i and using $index instead.